designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -58,24 +54,18 @@ trait DeploymentService extends ProcessStateService {
 
 trait ProcessStateService {
 
-  def fetchProcessStatesForProcesses(processes: List[BaseProcessDetails[Unit]])(
-      implicit user: LoggedUser,
-      ec: ExecutionContext,
-      freshnessPolicy: DataFreshnessPolicy
-  ): Future[Map[String, ProcessState]]
-
-  def enrichDetailsWithProcessState[T](processList: List[BaseProcessDetails[T]])(
+  def enrichDetailsWithProcessState[F[_]: Traverse](processTraverse: F[ScenarioWithDetails])(
       implicit user: LoggedUser,
       ec: ExecutionContext,
       freshnessPolicy: DataFreshnessPolicy
-  ): Future[List[BaseProcessDetails[T]]]
+  ): Future[F[ScenarioWithDetails]]
 
   def getProcessState(
       processIdWithName: ProcessIdWithName
   )(implicit user: LoggedUser, ec: ExecutionContext, freshnessPolicy: DataFreshnessPolicy): Future[ProcessState]
 
   def getProcessState(
-      processDetails: BaseProcessDetails[_]
+      processDetails: RepositoryScenarioWithDetails[_]
```

Comment:
Ehh thought that we could pass here just `ScenarioWithDetails`, but right now I see it's not so easy ;)

Reply:
I also wanted to do this, but we need to do more reactors to make it possible